File: backend/schemas/book_parts.py
from datetime import datetime
import uuid
from pydantic import BaseModel


class LightBookPartResponseSchema(BaseModel):
    id: uuid.UUID
    book_id: uuid.UUID
    parent_id: uuid.UUID | None
    label: str
    sibling_index: int
    is_story_part: bool
    is_entity_extracted: bool
    created_at: datetime
    level: int


class BookPartResponseSchema(LightBookPartResponseSchema):
    content: str


class BookPartUpdateSchema(BaseModel):
    is_story_part: bool

File: frontend/src/types/book_parts.ts
export interface LightBookPartResponseSchema {
    id: string;
    book_id: string;
    parent_id: string | null;
    label: string;
    sibling_index: number;
    is_story_part: boolean;
    is_entity_extracted: boolean;
    created_at: Date;
    level: number;
}

export interface BookPartResponseSchema extends LightBookPartResponseSchema {
    content: string;
}

export interface BookPartUpdateSchema {
    is_story_part: boolean;
}
